CorsixTH/Lua/hospital.lua

```diff
@@ -1987,6 +2037,121 @@ function Hospital:checkDiseaseRequirements(disease)
+    if nw ~= 0 then
+      walls[#walls + 1] = {wall = "south", parcel = flags.parcelId}
+    end
+  end
+
+  return walls
+end
+
+--! Test for the given position to be inside the given rectangle.
+--!param x (int) X position to test.
+--!param y (int) Y position to test.
+--!param rect (table x, y, width, height) Rectangle to check against.
+--!return (bool) Whether the position is inside the rectangle.
+local function isInside(x, y, rect)
+  return x >= rect.x and x < rect.x + rect.width and y >= rect.y and y < rect.y + rect.height
+end
+
+--! Find all ratholes that match the `to_match` criteria.
+--!param holes (list ratholes) Currently existing holes.
+--!param to_match (table) For each direction a rectangle with matching tile positions.
+--!return (list) Matching ratholes.
+local function findMatchingRatholes(holes, to_match)
+  local matched = {}
+  for _, hole in ipairs(holes) do
+    if isInside(hole.x, hole.y, to_match[hole.wall]) then table.insert(matched, hole) end
+  end
+  return matched
+end
+
+--! Remove the ratholes that use the walls of the provided room.
+--!param room (Room) Room being de-activated.
+function Hospital:removeRatholesAroundRoom(room)
+  local above_rect = {x = room.x, width = room.width, y = room.y - 1,          height = 1}
+  local below_rect = {x = room.x, width = room.width, y = room.y +room.height, height = 1}
+  local left_rect  = {x = room.x - 1,          width = 1, y = room.y, height = room.height}
+  local right_rect = {x = room.x + room.width, width = 1, y = room.y, height = room.height}
+
+  local to_delete = {east = left_rect, west = right_rect, south = above_rect, north = below_rect}
+
+  local remove_holes = findMatchingRatholes(self.ratholes, to_delete)
+  for _, hole in ipairs(remove_holes) do self:removeRathole(hole) end
+end
+
+-- Add a rathole to the room.
+--!param x (int) X position of the tile containing the rathole.
+--!param y (int) Y position of the tile containing the rathole.
+--!param wall (string) Wall containing the hole (north, west, south, east)
+--!param parcel (int) Parcel number of the xy position.
+function Hospital:addRathole(x, y, wall, parcel)
+  for _, rathole in ipairs(self.ratholes) do
+    if rathole.x == x and rathole.y == y and rathole.wall == wall then return end
+  end
+
+  local hole = {x=x, y=y, wall=wall, parcel=parcel}
+  if wall == "north" or wall == "west" then
+    -- Only add a rat-hole graphics object for the visible holes.
+    hole["object"] = self.world:newObject("rathole", hole.x, hole.y, hole.wall)
+  end
+  table.insert(self.ratholes, hole)
+end
+
+--! Remove the provided rathole.
+--!param hole (table{x, y, wall, optional object}) Hole to remove.
+function Hospital:removeRathole(hole)
+  for i, rathole in ipairs(self.ratholes) do
+    if rathole.x == hole.x and rathole.y == hole.y and rathole.wall == hole.wall then
+      table.remove(self.ratholes, i)
+      if rathole.object then self.world:destroyEntity(rathole.object) end
+      break
```

Comment:
Maybe an extra newline, or move the body of the if onto the next line. Just trying to separate the break so it doesn't look like it might belong with the previous if.